### ISSUE TITLE
use ActionChains to clear an input instead of `.clear()`

### ISSUE THREAD

**Issue by dskard** [yashaka/selene#188]
Sometimes Selenium's `.clear()` method doesn't work to delete the text in an input element.

In the element.py `.set()` method (https://github.com/yashaka/selene/blob/master/selene/elements.py#L353), perhaps we should be using ActionChains to clear inputs before we type into them, as shown in https://github.com/SeleniumHQ/selenium/issues/6741

**Comment by SergeyPirogov** [comment]
@dskard thank you, but it’s a very specific case! So I think for such situations user is able to use direct selenium calls

**Comment by dskard** [comment]
I would agree that the original post was very specific, but the solution seems to be the more general solution that people should be aiming to implement. From that conversation, my understanding is that the `.clear()` method will only work for a subset of input fields a developer may encounter. The ActionChains solution more closely models human behavior, which is what allows it to work on input fields where the `.clear()` method is currently failing.

If we stick with a `.set()` method that relies on the `.clear()` method under the hood, we force developers using Selene to write their own ActionChains code to clear the input and then use a `.send_keys()` call to populate the input. Doing this for one input is probably ok. Doing this for a form with multiple inputs would get verbose. My guess is people would prefer the concise `.set()` method that reliably clears the input field before typing into it.

Does that match with your understanding of the conversation or are there other details I am missing?

**Comment by aleksandr-kotlyar** [comment]
Closing this as stale / addressed by a workaround in Selene 2.x.

The exact proposal from this issue was not applied directly: `Element.set_value(...)` / `Element.set(...)` still use Selenium `webelement.clear()` followed by `send_keys(...)`.

However, Selene has a supported workaround for this case since `2.0.0rc2`:

```python
from selene import browser, command

browser.element('#field').perform(command.select_all).type('new value')
````

This was added specifically for cases where `element.set_value(text)` does not work well because it relies on `clear()`. Since `2.0.0rc9`, `command.select_all` is implemented via Selenium `ActionChains` and works both on element and browser level.

If there is still a need to replace the internals of `set_value(...)` itself, please open a new issue with a current minimal reproducible example.
